### PR TITLE
New document theme!

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

- Migrate documents to new theme.
- Reconstruct the documents structure.
- Added new instructions.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: docs/requirements.txt
Comment (jambayk) [inline docs/requirements.txt:1]: I thought Sam originally wanted to migrate from sphinx to mkdocs. Was that idea abandoned and you found a new theme instead?

Comment (xiaoyu-work) [reply docs/requirements.txt:1]: It needs more work on auto generated doc for code on Mkdocs, like CLI and Pass than Sphinx. I confirmed with Sam so what he wants is a clearer layout and make it easier for users to locate the How-to section in our documentation. So I found this theme that looks almost same as Mkdocs. Sam said he is okay with this theme.

Comment (jambayk) [reply docs/requirements.txt:1]: I see. Thanks for the info

File: .azure_pipelines/job_templates/olive-build-doc-template.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     steps:
     - template: olive-setup-template.yaml
       parameters:
-        python_version: '3.8'
+        python_version: '3.10'
```
Comment (jambayk) [inline .azure_pipelines/job_templates/olive-build-doc-template.yaml:20]: do the new dependencies require python 3.10? When i tried moving the doc build over to python 3.8, there was an issue with one of the dependencies. it's the same error in the failed doc build ci here

File: docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md [hunk 2]
```diff
@@ -28,8 +28,12 @@ Olive uses the `model_type` for the HuggingFace model and divides the transforme
 
 The following shows the final split models:
 
-<img align="center" alt="num_splits" width="500px" src="../images/model_splitting/num_splits.png"><br>
+```{figure} ../../../images/model_splitting/num_splits.png
+:width: 500px
+:align: center
 
+Olive Architecture
```
Comment (jambayk) [inline docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md:35]: left over from copy paste?

File: docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md
Comment (jambayk) [inline docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md:1]: this file might be better one level up. All of the other files under `model-opt-and-transform` deal with model type specific passes. but this is more like a blog on how to do model splitting.

Comment (xiaoyu-work) [reply docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md:1]: We probably should create a new folder specifically for techs blog to introduce some new techs @samuel100 . I'll move this one level up for now.

File: docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md (hunk 2 above)
Comment (xiaoyu-work) [reply docs/source/how-to/configure-workflows/model-opt-and-transform/model-splitting.md]: Updated

File: examples/getting_started/olive_quickstart.ipynb [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+   "metadata": {
+    "id": "MEfAinRUdxmG"
+   },
+   "source": [
+    "# ✨  OLIVE: Quickstart\n",
+    "\n",
+    "This notebook shows you how to get started with OLIVE. You will:\n",
+    "\n",
+    "1. use the `auto-opt` Olive CLI command to optimize an SLM model for the ONNX Runtime (for CPU devices).\n",
+    "1. Use the ONNX Runtime Python binding to execute a simple chat interace that consumes the model.\n",
```
Comment (jambayk) [inline examples/getting_started/olive_quickstart.ipynb:14]: interace -> interface typo

File: docs/source/how-to/cli/cli-finetune.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    --text_template "<|start_header_id|>user<|end_header_id|>\n{phrase}<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n{tone}" \
+    --method lora \
+    --max_steps 100 \
+    --log_level 1 \
+    --resource_group {RESOURCE_GROUP_NAME} \
+    --workspace_name {WORKSPACE_NAME} \
+    --aml_compute {COMPUTE_NAME}
+```
+
+You can download the model artefact using the Azure ML CLI:
```
Comment (jambayk) [inline docs/source/how-to/cli/cli-finetune.md:49]: spell checking complains about artefact vs artifact.

**Review by jambayk — APPROVED** [review]
thanks! LGTM as long as Sam is happy with it too.

**Pull request merged**